app.py

```diff
@@ -186,25 +185,29 @@ class MQStatus(object):
             logger.error("handle_file(): unable to decode msg as json: {}".format(msg.value))
             continue
 
-        hash_ = data['hash']
+        if data.get('payload_id'):
+            payload_id = data['payload_id']
+        else:
+            payload_id = data.get('hash')
```

Comment:
for backward compatibility. If a validator does not update to use `payload_id` then we still want to be able to accept `hash`. We can pull this out once everyone is updated and we don't see this anymore.

Reply:
What if the payload doesn’t contain neither _payload_id_, nor _hash_? It’s still data from an external service. Do we just want pass _None_ to _storage.copy_? Maybe it’d be good to raise an exception instead.

These four lines can be contracted to a following one line. I am not sure whether it’s more readable though.

```python
payload_id = data.get('payload_id', data.get('hash'))
```

Even if not, I’d find the following version more idiomatic:

```python
if 'payload_id' in data:
    payload_id = data['payload_id']
else:
    payload_id = data.get('hash')
```

or

```python
payload_id = data['payload_id'] if 'payload_id' in data else data.get('hash')
```